asm/io.asm

```diff
@@ -137,6 +127,55 @@ REFILL ; ( -- flag )
     dec $d020
     jmp -
 
+.return_false
+    dex
+    lda #0
+    sta MSB,x
+    sta LSB,x
+    rts
+
+.getLineFromEvaluateString
+    lda EVALUATE_STRING_SIZE_LSB
+    ora EVALUATE_STRING_SIZE_MSB
+    beq .return_false
+
+EVALUATE_STRING_PTR_LSB = * + 1
+    lda #0
+    sta TIB_PTR
+EVALUATE_STRING_PTR_MSB = * + 1
+    lda #0
+    sta TIB_PTR + 1
+
+.grow_tib_to_end_of_line
+    lda EVALUATE_STRING_PTR_LSB
+    sta + + 1
+    lda EVALUATE_STRING_PTR_MSB
+    sta + + 2
++   lda PLACEHOLDER_ADDRESS
+    tay
+
+    inc EVALUATE_STRING_PTR_LSB
+    bne +
+    inc EVALUATE_STRING_PTR_MSB
++
+    lda EVALUATE_STRING_SIZE_LSB
+    bne +
+    dec EVALUATE_STRING_SIZE_MSB
++   dec EVALUATE_STRING_SIZE_LSB
+
+    tya
+    cmp #$d
+    beq .return_true
+
+    inc TIB_SIZE ; assumes max line length = 255
```

Comment:
... this is an assumption not in standard EVALUATE, and it's one that I think breaks my own personal BLOCK wordset that defines `: load block 1024 evaluate ;`. Maybe it's better to have a separate word after all :sweat:

Reply:
fixed.